R/get_analysis_date.R

```diff
@@ -0,0 +1,270 @@
+      }
+    }
+  }
+  input_check_scale(planned_calendar_time, label = "planned_calendar_time")
+  input_check_scale(target_event_overall, label = "target_event_overall")
+  input_check_scale(max_extension_for_target_event, label = "max_extension_for_target_event")
+  input_check_scale(min_time_after_previous_analysis, label = "min_time_after_previous_analysis")
+  input_check_scale(min_n_overall, label = "min_n_overall")
+  input_check_scale(min_followup, label = "min_followup")
+
+  input_check_vector <- function(x = NA, label = "x"){
+    if (!(all(is.na(x) | (is.numeric(x) & x > 0)))) {
+      stop(paste0(label, " must be a positive number with either NA or positive numbers!"))
+    }
+  }
+  input_check_vector(target_event_per_stratum)
+  input_check_vector(min_n_per_stratum)
+
+  cond1 <- inherits(enroll_rate, c("tbl_df", "data.frame")) # check if enrollment is input by user
+  cond2 <- !is.na(min_n_overall)                            # check if min_n_overall is input by user
+  cond3 <- !all(is.na(min_n_overall))                       # check if min_n_per_stratum is input by user
+  if(cond1){
+    n_max <- sum(enroll_rate$rate * enroll_rate$duration)
+
+    if(is.na(min_followup)){
+      stop("min_followup must be provided!")
+    }
+
+    if(cond2){
+      if(min_n_overall > n_max){
+        stop("min_n_overall should be a positive number smaller than the total sample size!")
+      }
+    }
+
+    if(cond3){
+      if(sum(min_n_per_stratum, na.rm = TRUE) > n_max){
+        stop("min_n_per_stratum should be a sum of positive numbers smaller than the total sample size!")
+      }
+    }
+  }
+
+  # cutting option 1: planned calendar time for the analysis
+  cut_date1 <- planned_calendar_time
+
+  # cutting option 2:  reach targeted events
+  # 2a: reach targeted events of the overall population
+  if(!is.na(target_event_overall)){
+    cut_date2a <- get_cut_date_by_event(simu_data, event = target_event_overall)
+  } else {
+    cut_date2a <- NA
+  }
+  # 2b: reach targeted events per sub-population
+  if(!all(is.na(target_event_per_stratum))){
+    stratum <- unique(simu_data$stratum)
+    cut_date2b <- lapply(seq_along(target_event_per_stratum),
+                         function(x){
+                           get_cut_date_by_event(simu_data %>% dplyr::filter(stratum == stratum[x]),
+                                                 event = target_event_per_stratum[x])
+                         }) %>% unlist() %>% max()
+  } else {
+    cut_date2b <- NA
+  }
+  cut_date2 <- pmax(cut_date2a, cut_date2b, na.rm = TRUE)
+
+  # cutting option 3: max time extension to reach targeted events
+  cut_date3 <- max_extension_for_target_event
+
+  # cutting option 4: planned minimum time after the previous analysis
+  cut_date4 <- previous_analysis_date + min_time_after_previous_analysis
+
+  # cutting option 5: minimal follow-up time after specified enrollment fraction
+  get_min_date <- function(enroll_rate, min_n = 400) {
```

Comment:
As above, please move this function to the bottom of the file

Reply:
Echo with my reply to the 1st comment.